### PR TITLE
Npm: Raise helpful err for forbidden missing deps

### PR DESCRIPTION
**Pull request by feelepxyz**
Raise a helpful error when the default registry is private and doesn't
include scoped public packages. We already do this for yarn.

Fixes: https://sentry.io/dependabot/backend/issues/797560387/

### CODE REVIEW

**Comment by feelepxyz** [comment]
@greysteil haven't added any specs, had a go at this a while back when looking at the one for yarn, but it was such a pain to get the npmrc setup working.. *might* have a go later but didn't want to block this. Also currently handling a sentry for one user!

**Comment by greysteil** [comment]
Nice. Can we get a test? Just make up a bullshit scope...

**Comment by feelepxyz** [comment]
@greysteil oh yeah guess we don't need to replicate the whole setup!

**Comment by feelepxyz** [comment]
@greysteil added some specs ✌️

**Comment by greysteil** [comment]
Fixture and test look 👌

File: spec/dependabot/file_updaters/java_script/npm_and_yarn_spec.rb [hunk 1]
```diff
@@ -1502,10 +1564,17 @@
...
+          "github_sub_dependency_version_missing.json"
+        end
+        let(:npm_lock_fixture_name) do
+          "github_sub_dependency_version_missing.json"
+        end
+        let(:yarn_lock_fixture_name) do
+          "github_sub_dependency_version_missing.lock"
+        end
 
+        # TODO: rename this to test-missing-dep-version-npm-package
```
Comment (greysteil) [inline spec/dependabot/file_updaters/java_script/npm_and_yarn_spec.rb]: You OK to take this @feelepxyz ?

Comment (feelepxyz) [reply spec/dependabot/file_updaters/java_script/npm_and_yarn_spec.rb]: Yup!

Comment (feelepxyz) [reply spec/dependabot/file_updaters/java_script/npm_and_yarn_spec.rb]: Done

**Pull request merged**